tools/main.cpp

```diff
@@ -26,13 +26,15 @@ static V depends(V&& v) { return std::forward<V>(v); }
 template <typename... Ts>
 struct runner {
    explicit runner(CLI::App& app)
-      : tup(depends<Ts>(app)...) {}
+      : tup(depends<Ts>(app)...), _app(app) {}
 
    template <std::size_t I=0>
    constexpr inline int exec() {
       if constexpr (I == sizeof...(Ts)) {
-         std::cerr << "Please run one of the subcommands available. Use --help to see what is available." << std::endl;
-         return -1;
+         std::cout << _app.help() << std::endl

```

Comment:
```suggestion
         std::cerr << _app.help() << std::endl
```

Reply:
`git` prints help to stdout and exits with error. We can and should do the same.

Reply:
Done.
I'm returning error `1`. For negative return codes in other places I have created an issue: https://github.com/AntelopeIO/antler-proj/issues/57